### ISSUE TITLE
Support private github repos

### ISSUE THREAD

**Issue by giggio** [Hacklone/private-bower#111]
I am trying to hit an https git repo. It requires an auth header (`Authorization: Basic somebase64auth`). I can't get private-bower to register the package with something like this: `bower register package git://myserver.com/projects/_git/package`. It fails with:

````
Failed to execute "git ls-remote --tags --heads git://myserver.com/projects/_git/package", exit code of #128
````

If I try with `bower register package https://myserver.com/projects/_git/package` it succeeds, but later calls fail when I try to install `package`.

What is the correct way to register a package that demands the `Authorization` header, through https?

**Comment by Hacklone** [comment]
Hey, 

You can permanently store the https credentials (https://confluence.atlassian.com/display/STASH/Permanently+authenticating+with+Git+repositories)
ot just simply use ssh instead of https with a key.

This is not an issue of the private-bower, so I'll close this now but I'm happy to help if you do not succeed :)

**Comment by giggio** [comment]
Ok, I was able to get it to work. I used a .netrc file on the server and it was able to cache the login info. I also had to use a replacement on git urls on the server so that it is able to get to my server:

````
[url "http://myserver.cloudapp.net/"]
	insteadOf = git://myserver.cloudapp.net/
````

I also had to that on my local client, but only to register packages. As I enabled a cache with git, now private-bower is serving the files as well.

I believe this could be a quite common scenario, so I am leaving it documented here. Thanks for the support!
